src/styles/main.scss

```diff
@@ -1,3 +1,68 @@
-body {
-  margin: 0;
+$month-length:
+  28,
+  29,
+  30,
+  31,
+;
+$size: 100px;
+$gap: 1px;
+$padding: 10px;
+$days:
+  "mon" 0,
+  "tue" 1,
+  "wed" 2,
+  "thu" 3,
+  "fri" 4,
+  "sat" 5,
+  "sun" 6,
+;
+
+@each $length in $month-length {
+  .cal__day:nth-child(#{$length + 1}) {
+    display: none;
+  }
+}
+
+@each $name, $index in $days {
+  .cal__day:first-child {
+    margin-left: $size * $index;
+  }
+}
+
+.cal {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($size + $gap) * 7 + $padding * 2;
+  gap: $gap;
+  padding: $padding;
+  margin: 0 auto;
+
+  @for $i from 1 through 31 {
+    &__day:nth-child(#{$i})::before {
+      content: "#{$i}";
+      margin: auto;
+    }
+  }
+
+  &__day {
+    gap: $gap;
+    cursor: pointer;
+    box-sizing: border-box;
+    display: flex;
+    flex-direction: column;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    color: black;
+    background-color: #eee;
+    width: $size - 1px;
+    height: $size - 1px;
+    border: 1px solid black;
+    margin: 0;
+    transition: all 0.5s;
+  }
+
+  &__day:hover {
+    background-color: pink;
+    transform: translateY(-20px);
+  }
```

Comment:
1. move to separate file - blocks/calendar.scss
2. transition should have every property written, not all `transition: background-color 0.5s, transform 0.5s` or `transition-property: backgroun-color, transform; transition-duration: 0.5s;`
3. something is wrong if you need to subtract 1px from size
4. you can define font-family and all font settings for the whole calendar or even body, not for each day
5. it's a good practice to have `box-sizing: border-box` for each element, use `*` selector for it 
6. each day can have justify-content and align-items center, so you don't need to set margin auto for `:before`